api/catchpoint.go

```diff
@@ -0,0 +1,73 @@
+package api
+
+import (
+	"errors"
+	"io"
+	"net/http"
+	"strings"
+)
+
+type CatchPointUrl string
+
+const (
+	FNet    CatchPointUrl = "https://fnet-catchpoints.algorand.green/latest"
+	BetaNet CatchPointUrl = "https://algorand-catchpoints.s3.us-east-2.amazonaws.com/channel/betanet/latest.catchpoint"
+	TestNet CatchPointUrl = "https://algorand-catchpoints.s3.us-east-2.amazonaws.com/channel/testnet/latest.catchpoint"
+	MainNet CatchPointUrl = "https://algorand-catchpoints.s3.us-east-2.amazonaws.com/channel/mainnet/latest.catchpoint"
+)
+
+const InvalidNetworkParamMsg = "invalid network"
+
+type LatestCatchpointResponse struct {
+	HTTPResponse   *http.Response
+	ResponseCode   int
+	ResponseStatus string
+	JSON200        string
+}
+
+func (r LatestCatchpointResponse) StatusCode() int {
+	return r.ResponseCode
+}
+func (r LatestCatchpointResponse) Status() string {
+	return r.ResponseStatus
+}
+
+func GetLatestCatchpointWithResponse(http HttpPkgInterface, network string) (LatestCatchpointResponse, error) {
+	var response LatestCatchpointResponse
+
+	var url CatchPointUrl
+	switch network {
+	case "fnet-v1", "fnet":
+		url = FNet
+	case "betanet-v1.0", "betanet":
+		url = BetaNet
+	case "testnet-v1.0", "testnet":
+		url = TestNet
+	case "mainnet-v1.0", "mainnet":
+		url = MainNet
+	}
+	if url == "" {
+		return response, errors.New(InvalidNetworkParamMsg)
+	}
+	res, err := http.Get(string(url))
+	response.HTTPResponse = res
+	if err != nil {
+		return response, err
+	}
+	defer res.Body.Close()
```

Comment:
should this be on line 64 instead?

Reply:
Request body _should_ be present on all non-nil errors.